docs/src/doc/contribution/knowledge-base/entry-generation.md

```diff
@@ -2,35 +2,30 @@
+## The godot-class-graph-symbol-processor
 
-Previously, we implemented our own compiler plugin based on the [mpapt](https://github.com/Foso/MpApt) project.
+Previously, we implemented our own compiler plugin based on the [mpapt](https://github.com/Foso/MpApt) and [ksp](https://github.com/google/ksp) project.
 Even though it worked very well, the code was far from easy to understand and as the compiler plugin API is not stable yet.
-Furthermore, `MpApt` introduced a lot of maintenance work to keep it up to date with the ever evolving Kotlin compiler.
+Furthermore, `MpApt` introduced a lot of maintenance work to keep it up to date with the ever evolving Kotlin compiler and `Ksp` wasn't able to process Scala files.
 
-To mitigate that problem we switched to [KSP (Kotlin Symbol Processing API)](https://github.com/google/ksp) which is basically a Kotlin Annotation Processor (KAPT) on steroids.
-Meaning it in essence is a compiler plugin which lets you inspect the source code being compiled through a abstracted and simplified API.
-This means for us that we can work against a more stable API and the work of keeping it up to date with Kotlin compiler changes is offloaded
-to a well maintained open source project with far more people with better knowledge about the internal workings of the Kotlin compiler.
+To mitigate that problem we switched to [ClassGraph](https://github.com/classgraph/classgraph) which is Bytecode processor.
+Meaning it in essence is a compiler plugin which lets you inspect the compiled bytecode with an abstracted and simplified API.
+This means for us that we can work against a more stable API and the work of keeping it up to date with the Kotlin language is removed.
 
-We use our implementation of a Kotlin Symbol Processing to gather all needed information for the entry generator and convert it into the format it understands.
-With this it should be more easy to implement additional JVM languages as the entry generator itself can stay the same for all languages. Only the symbol processor has to change per language.
-
-## Default Value Extraction
-
-We extract the default values with the help of JetBrains' [Program Structure Interface (PSI)](https://plugins.jetbrains.com/docs/intellij/psi.html),
-that is a language specific AST (Abstract Syntax Tree). But as the simplified API of KSP does not allow the introspection of the source code representation,
-we implemented this ourselves in the `godot-kotlin-symbol-processor` as well.
+We use ClassGraph to gather all needed information for the entry generator and convert it into the format it understands.
+With this it should be far easier to implement additional JVM languages as ClassGraph is language agnostic and only minor tweak are necessary to support additional static languages.
 
 ## Debugging
 
-As `godot-kotlin-symbol-processor` is a Kotlin compiler plugin and thus the execution context of the entry generator is as well,
+TODO: Write the new way to debug the bytecode processing
```

Comment:
<img width="1154" alt="image" src="https://github.com/user-attachments/assets/3eb58428-5a8d-4069-8ae7-a765943eb31a" />
You can debug normally as you would debug normal gradle tasks.

also classgraph is no compiler plugin anymore. we just do two full compilation rounds and using incremental compilation to our advantage.

But it's ok if you do not want to update this part in this PR. I can also update the documentation here in a separate PR if you prefer.